activerecord/lib/arel/table.rb

```diff
@@ -10,11 +10,15 @@ class << self; attr_accessor :engine; end
 
     attr_accessor :name, :table_alias
 
-    # TableAlias and Table both have a #table_name which is the name of the underlying table
-    alias :table_name :name
-
     def initialize(name, as: nil, klass: nil, type_caster: klass&.type_caster)
-      @name = name.to_s
+      @name =
+        case name
+        when Symbol then name.to_s
```

Comment:
To preserve current behavior (covered with tests). Not sure it leaked outside of tests though, but prefer to be defensive here.